### PR TITLE
Styrspak ljudeffekter

### LINKED ISSUE #86: Add sound effects to the Tank controllers
As a gamer, I want to hear music when I move the joystick and play the game so that I have fun playing the game.

Acceptance criteria:
- [x] Add sound effect to the buttons and joystick
- [x] Add a proper ear-friendly music to the background of game

### PR DESCRIPTION
**Pull request by Spidergirl93**
Sound effects for the new manual screen and joystick has been added. However they are not perfect and can be replaced with higher quality sounds in the future. 
I had to change the setUpJoystick method for a better implementation of its sound effects. It' basically the same as before only added a tiny onTouch() method to it. Check it out and let me know there is a problem.
Also I added another class to utilities for a better sound effect handling and cleaner code body.
P.S. I also changed the delay time of the joystick to 30 in setUpJoystick() since I could feel a delay while controlling the car. I believe now the car reacts faster when I use joystick.

### CODE REVIEW

**Review by hannahshiels — COMMENTED** [review]
Could you just delete the inital setupJoystick method? Naming something setUpJoystick1 is kind of unclear what it means, and the other code is not necessary anymore 🤔

**Comment by Spidergirl93** [comment]
> Could you just delete the inital setupJoystick method? Naming something setUpJoystick1 is kind of unclear what it means, and the other code is not necessary anymore 🤔

Yes I agree. I will fix the naming after you guys approve the body of the method. Just didn't want to replace the last one since someone else wrote it and it's kinda rude :)

File: androidGeoBot/app/src/main/java/com/example/androidtank/ManualActivity.java [hunk 1]
```diff
@@ -106,32 +111,53 @@ public void setTankControls () {
...
-            public void onClick(View view) {
-                client.button_publish(button);
+            public boolean onTouch(View v, MotionEvent event) {
+                switch(event.getAction()) {
+                    case MotionEvent.ACTION_DOWN:
+                        if (mHandler != null) return true;
+                        mHandler = new Handler();
+                        mHandler.postDelayed(mAction, 100);
+                        effects.startEffect(ManualActivity.this, R.raw.break_sound,
+                                0.9f,false);
```
Comment (lucasnordic) [inline androidGeoBot/app/src/main/java/com/example/androidtank/ManualActivity.java]: 0.5f felt better here

File: androidGeoBot/app/src/main/java/com/example/androidtank/ManualActivity.java [hunk 2]
```diff
@@ -140,17 +166,45 @@ public void onClick(View view) {
...
+        joystick = (JoystickView) findViewById(R.id.joystickView);
+        joystick.setOnTouchListener(new JoystickView.OnTouchListener() {
             @Override
-            public void onMove(int angle, int strength) {
-                int newX = convertJoystickX(); // for determining angle strength
-                client.joystick_publish(joystick, angle, strength, newX);
+            public boolean onTouch(View v, MotionEvent event) {
+                if (event.getAction() == event.ACTION_DOWN) {
+
+                    effects.startEffect(ManualActivity.this, R.raw.acceleration, 0.4f, true);
```
Comment (lucasnordic) [inline androidGeoBot/app/src/main/java/com/example/androidtank/ManualActivity.java]: 0.2f felt better here

File: androidGeoBot/app/src/main/java/com/example/androidtank/ManualActivity.java [hunk 3]
```diff
@@ -140,17 +166,45 @@ public void onClick(View view) {
...
+                    int delay = 30;
+                    joystick.setOnMoveListener(new JoystickView.OnMoveListener() {
+                        public void onMove(int angle, int strength) {
+                            int newX = convertJoystickX(); // for determining angle strength
+                            client.joystick_publish(joystick, angle, strength, newX);
+                        }
+                    }, delay);
+                } else {
+                    //TODO find a better sound effect. This is too annoying and low!
+                    effects.startEffect(ManualActivity.this, R.raw.carsound, 0.9f, false);
```
Comment (lucasnordic) [inline androidGeoBot/app/src/main/java/com/example/androidtank/ManualActivity.java]: 1.0f felt better here

**Comment by lucasnordic** [comment]
Is there anything we can do about the car acceleration sound repeating? Is it possible to have something like, once the acceleration is nearing the end and we want to start playing it again, we can start from a specific place in the sound-clip instead? I was thinking at around 9-10 seconds into the soundclip, so it doesn't start from the beginning.

**Comment by Spidergirl93** [comment]
> Is there anything we can do about the car acceleration sound repeating? Is it possible to have something like, once the acceleration is nearing the end and we want to start playing it again, we can start from a specific place in the sound-clip instead? I was thinking at around 9-10 seconds into the soundclip, so it doesn't start from the beginning.

That's a good point. I'll look into it

**Review by hannahshiels — APPROVED** [review]
From what I've seen (well mostly heard 😃), its working great now.

**Pull request merged**